### ISSUE TITLE
Use white/black icon/text depending upon chosen color to ensure visibility

### ISSUE THREAD

**Issue by Riduidel** [stuartcrobinson/unique-window-colors#15]
Currently, this plugin provides a nice set of distinct colors.
But both menu bar texts and left panel icons may sometimes be impossible to read.

Take a look at this example

![image](https://user-images.githubusercontent.com/171018/55466317-3e320e00-55ff-11e9-9951-b96e3dd39ad8.png)

Icons are somewhat visible, but menu text is really hard to read (unless hovered).
A typical solution is to check color darkness, and swap text/icon drawing color from white to black to ensure visibility. (and I guess that's what [peacock](https://github.com/johnpapa/vscode-peacock) does).

**Comment by stuartcrobinson** [comment]
ah thank you for letting me know!  hadn't seen this before.  

could you send me:

1.  the contents of your `.vscode/settings.json` file
2. the Window Colors settings you have currently enabled, and 
3. any info on what color theme you're currently using?  

so i can reproduce.

**Comment by Riduidel** [comment]
Content of `.vscode/settings.json` is

```
{
    "workbench.colorCustomizations": {
        "activityBar.background": "#695641",
        "titleBar.activeBackground": "#FFFFFF",
        "titleBar.activeForeground": "#FFFFFF"
    }
}
```

By windows color settings, I guess you mean the unique windows color settings section in my main settings ? If so, here they are

```
"windowColors.🌈 BaseColor": ""
"windowColors.🌈 DeleteSettingsFileUponExit": false
"windowColors.🌈 Theme": "dark"
```

I wonder what the "theme" settings configure ...

I'm using as color theme 
```
"workbench.colorTheme": "Solarized Light"
```

**Comment by Riduidel** [comment]
OK, I've understood the goal of the `"windowColors.🌈 Theme": "dark"`. if I set it to `"light"` text/icons are visible. So I guess the best case would be to auto-toggle that value according to selected VSCode theme, no ?

**Comment by stuartcrobinson** [comment]
yeah, the `windowColors.🌈 Theme` setting determines whether light or dark colors are used.  but the
 foreground/font color should be generated to contrast nicely with the calculated background color.

i'm on mac and haven't seen this happen before -- i'll test on windows within the next few days

**Comment by stuartcrobinson** [comment]
Hi Nicolas, I just published a revision to how colors are generated and can be selected.  Feel free to let me know what you think -- if this helps font vs background colors for you?

**Comment by stuartcrobinson** [comment]
false alarm.  new version had trouble in windows so i've reverted for now.
